pallets/sponsorship/src/migration.rs

```diff
@@ -0,0 +1,216 @@
+/*
+ * This file is part of the Nodle Chain distributed at https://github.com/NodleCode/chain
+ * Copyright (C) 2020-2022  Nodle International
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+use crate::{BalanceOf, Config, Pallet, Pot, User, UserRegistrationCount};
+use codec::{Decode, Encode};
+use frame_support::{
+	traits::{Get, StorageVersion},
+	weights::Weight,
+};
+use support::LimitedBalance;
+
+/// The current storage version.
+pub const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
+mod v0 {
+	use super::*;
+	#[derive(Encode, Decode, Debug)]
+	pub struct PotDetails<AccountId, SponsorshipType, Balance: frame_support::traits::tokens::Balance> {
+		pub sponsor: AccountId,
+		pub sponsorship_type: SponsorshipType,
+		pub fee_quota: LimitedBalance<Balance>,
+		pub reserve_quota: LimitedBalance<Balance>,
+	}
+
+	#[derive(Encode, Decode, Debug)]
+	pub struct UserDetails<AccountId, Balance: frame_support::traits::tokens::Balance> {
+		pub proxy: AccountId,
+		pub fee_quota: LimitedBalance<Balance>,
+		pub reserve_quota: LimitedBalance<Balance>,
+	}
+
+	pub type PotDetailsOf<T> =
+		PotDetails<<T as frame_system::Config>::AccountId, <T as Config>::SponsorshipType, BalanceOf<T>>;
+	pub type UserDetailsOf<T> = UserDetails<<T as frame_system::Config>::AccountId, BalanceOf<T>>;
+}
+
+/// Call this during the next runtime upgrade for this module.
+pub fn on_runtime_upgrade<T: Config>() -> Weight {
+	let mut weight: Weight = T::DbWeight::get().reads(1);
+
+	let translate_pot_details = |pre: v0::PotDetailsOf<T>| -> crate::PotDetailsOf<T> {
+		let mut fee_quota = LimitedBalance::with_limit(pre.fee_quota.limit());
+		fee_quota.saturating_add(pre.fee_quota.balance());
+		let mut reserve_quota = LimitedBalance::with_limit(pre.reserve_quota.limit());
+		reserve_quota.saturating_add(pre.reserve_quota.balance());
+		crate::PotDetailsOf::<T> {
+			sponsor: pre.sponsor,
+			sponsorship_type: pre.sponsorship_type,
+			fee_quota,
+			reserve_quota,
+			deposit: Default::default(),
```

Comment:
```suggestion
			deposit: Zero::zero(),
```

Probably more expressive than `Default`.